02-intermediate-ruby/references-and-values.md

```diff
@@ -0,0 +1,278 @@
+# References and Values
+
+## Learning Goals
+At the end of this lesson, students should be able to...
+
+- Discuss how objects are stored in Ruby
+- Differentiate between _references_ and _values_
+- Compare _modifying an object_ with _reassigning a variable_
+
+## Motivation
+
+We're going to start today with some Ruby code that does something a little unexpected. It's a method that takes an array of strings as an argument, and truncates (chops off the end of) all the strings with more than three characters. Or at least that's the idea.
+
+```ruby
+def short_strings(input)
```

Comment:
Since we specifically mention `+=` versus `push` later on I think this example code could be rewritten to use those methods, and be simplified at the same time:

Original:
```ruby
def add_llama(pets)
  pets += ['llama']
end

pets = ['dog', 'parrot', 'cat']
add_llama(pets)
puts pets
```

Fixed:
```ruby
def add_llama(pets)
  pets.push 'llama'
end

pets = ['dog', 'parrot', 'cat']
add_llama(pets)
puts pets
```

(Or use `<<` instead of `push`, whichever the students would be more familiar with at this point.)